src/features/members/MemberList.jsx

```diff
@@ -11,31 +13,56 @@ export default function MemberList({ cohort }) {
   }
 
   const memberData = cohortMembers.map(member => ({
+    id: member.id,
     email: member.email,
-    status: member.user ? 'Registered' : 'Invited',
+    status: COHORT_MEMBERSHIP_STATUS_TYPE_MAP[member.status],
+    active: member.active,
   }));
 
+  const statusFilterChoices = [
+    {
+      name: COHORT_MEMBERSHIP_STATUS_TYPE_MAP[COHORT_MEMBERSHIP_STATUS.activated],
+      value: COHORT_MEMBERSHIP_STATUS_TYPE_MAP[COHORT_MEMBERSHIP_STATUS.activated],
+    },
+    {
+      name: COHORT_MEMBERSHIP_STATUS_TYPE_MAP[COHORT_MEMBERSHIP_STATUS.invited],
+      value: COHORT_MEMBERSHIP_STATUS_TYPE_MAP[COHORT_MEMBERSHIP_STATUS.invited],
+    },
+    {
+      name: COHORT_MEMBERSHIP_STATUS_TYPE_MAP[COHORT_MEMBERSHIP_STATUS.deactivated],
+      value: COHORT_MEMBERSHIP_STATUS_TYPE_MAP[COHORT_MEMBERSHIP_STATUS.deactivated],
+    },
+  ];
+
   return (
     <section className="mt-3">
       <DataTable
         isFilterable
         isPaginated
         isSortable
+        initialState={{ hiddenColumns: ['id', 'active'] }}
         itemCount={cohortMembers.length}
         defaultColumnValues={{ Filter: TextFilter }}
         numBreakoutFilters={2}
         data={memberData}
         columns={[
+          { Header: 'Membership ID', accessor: 'id' },
           { Header: 'Email', accessor: 'email' },
+          { Header: 'Active', accessor: 'active' },
           {
             Header: 'Status',
             accessor: 'status',
-            Filter: DropdownFilter,
+            Filter: CheckboxFilter,
             filter: 'includesValue',
-            filterChoices: [
-              { name: 'Registered', value: 'Registered' },
-              { name: 'Invited', value: 'Invited' },
-            ],
+            filterChoices: statusFilterChoices,
+          },
+          {
+            Header: 'Action',
+            /* eslint-disable */
+            Cell: ({ row }) => (
+              <MemberActionRow member={row.values} cohort={cohort} />
+            ),
+            /* eslint-enable */
```

Comment:
ESLint flags this for recreating the component(s) on every render, but I don't know a way around that using the Paragon DataTable.

Reply:
🤔 I wonder if building a list of components along with the member data would do it?